File: simulate/66+1.cpp
#include <vector>
using namespace std;
class Solution {
    public:
        vector<int> plusOne(vector<int>& digits) {
            int c=1;
            int n=digits.size();
            for(int i=n-1;i>=0;i--){
                if(c==0)return digits;
                digits[i]+=c;
                c=0;
                if(digits[i]>=10){
                    c=1;
                    digits[i]%=10;
                }
            }
            if(c!=0)digits.insert(digits.begin(),c);
            return digits;
        }
    };

File: simulate/66+1.py
from typing import List
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        c=1
        for now in range(len(digits)-1,-1,-1):
            if c == 0: return digits
            digits[now] += c
            c = 0
            if digits[now] == 10:
                digits[now] = 0
                c = 1
        if c == 1:
            digits.insert(0,1)
        return digits